### DISCUSSION TITLE
Are the axes wrong on the confusion matrix?

### DISCUSSION THREAD

**Discussion by rurwin** [pycaret/pycaret#3924] [Q&A]
![image](https://github.com/pycaret/pycaret/assets/1812668/56401a17-0d3b-4a32-87bd-c0c515b8e1c5)

I read that as the 6 being a false positive on the confusion matrix, but a false negative on the error histogram.

Am I reading the charts incorrectly? (That is highly likely but I don't see how I can be.)
Otherwise the simplest explanation is that the confusion matrix has the true class on the X-axis and the predicted class on the Y axis.

The easiest way to see this is to run the PyCaret Binary Classification tutorial and plot an error histogram after the confusion matrix.
https://github.com/pycaret/pycaret/blob/master/tutorials/Tutorial%20-%20Binary%20Classification.ipynb

**Comment by imba-tjd** [comment]
See https://github.com/ultralytics/ultralytics/issues/2466#issuecomment-1542747344

unbelievable.

**Comment by rurwin** [comment]
> the "True" axis represents the predicted values of the model and the "Predicted" axis represents the labeled validation data.

I don't know about YOLOv8, but it isn't the case for PyCaret CM plots, as seen above in this thread.
It's also flipped 90 degrees. For a PyCaret plot, the Y axis is "True" and the X axis is "Predicted."

In PyCaret, if one plots the CM for various optimizers, the rows always add up to the same number, whereas the columns don't.
This is because the data labels are fixed. For example, the Diabetes training data has 150 negatives and 81 positives. The rows add up to 150 (True 0) and 81 (True 1).

My conclusion for this thread was that the Prediction Error plot was not showing what I thought it was. I didn't find a problem with the CM.

Looking at that YOLOv8 plot, the banding would seem to show that the columns were the model output. It doesn't make sense that the training data would have unrepresented classes. So the plots are oriented the same but labeled differently.

It isn't a pure CM though. The numbers have been converted to a fraction. That has lost information. The rows do not add up to 1 or any other constant value.

**Comment by imba-tjd** [comment]
I didn't use YOLO nor pycaret. I'm a beginner and trying to learn which labels for x and y axis are more common.

let's firstly talk about this thread.

> I didn't find a problem with the CM.

Do you mean, now, you think both of the picture are correct?

In the second picture, I see the number of "actual == 1" is 19. But in the first picture, "predicted == 1" is 19.

**Reply by rurwin** [reply]
I didn't find a problem with the CM here because I believe it is labelled correctly and shows the right data. I found the Class Prediction Error chart to be confusing. To my mind, its use of the term "actual" is misleading.
English is an ambiguous language . There is no right way to label these data. The only thing we can  do is to understand how people normally label them. We can disagree that it is the best way but it doesn't help anyone if we do something different.
In this case, we can see that the labelled data can be viewed as either the "true" categorization or the "predicted" output of the model. Whereas the model output is either the "predicted" values of the labels or the "actual" output of the model.
Different terminology suits different situations but chopping and changing just leads to confusion.
